### PR TITLE
Refactor Storage classes

### LINKED ISSUE #27: Remove redundant classes for Storage
Remove JsonAddressBook class and its dependents.

### LINKED ISSUE #275: Refactor JSON serialization and Storage classes
- [x] Refactor JSON serialization object classes and their test cases
  - [x] JsonSerializableBlockTree
  - [x] JsonSerializableBlockTreeItem
  - [x] JsonBlockStorage

- [x] Update `StorageManager` class and tests

### PR DESCRIPTION
**Pull request by firzanarmani**
Closes #27 and #275
- [x] Refactor JSON adapted object classes
    - [x] Update Block's `equals` conditions
- [x] Add tests for JSON adapted object classes
- [x] Update StorageManager and tests

### CODE REVIEW

File: src/main/java/com/notably/model/block/BlockImpl.java [hunk 1]
```diff
@@ -44,4 +44,17 @@ public Title getTitle() {
...
+        return other == this // short circuit if same object
+                || (other instanceof Block // instanceof handles nulls
+                && title.getText().equals(((Block) other).getTitle().getText())
+                && body.getText().equals(((Block) other).getBody().getText())); // state check
+    }
+
+    @Override
+    public int hashCode() {
+        return super.hashCode();
+    }
```
Comment (kevinputera) [inline src/main/java/com/notably/model/block/BlockImpl.java]: Let's do `Objects.hash(title.getText())` instead?

File: src/main/java/com/notably/model/block/BlockTreeImpl.java [hunk 2]
```diff
@@ -63,4 +63,24 @@ public void remove(AbsolutePath path) {
...
+        }
+
+        BlockTree otherTree = (BlockTree) other;
+
+        return this.getRootBlock().equals(otherTree.getRootBlock());
+    }
+
+    @Override
+    public int hashCode() {
+        return super.hashCode();
```
Comment (kevinputera) [inline src/main/java/com/notably/model/block/BlockTreeImpl.java]: Let's do `return getRootBlock().hashCode()` here?

File: src/main/java/com/notably/model/block/BlockTreeItemImpl.java [hunk 3]
```diff
@@ -132,12 +131,23 @@ public boolean equals(Object obj) {
...
+            }
+        }
+
+        return this.getTitle().getText().equals(otherBlock.getTitle().getText());
     }
 
     @Override
     public int hashCode() {
-        return hash(this.getTitle(), this.getTreeItem().getParent());
+        return super.hashCode();
```
Comment (kevinputera) [inline src/main/java/com/notably/model/block/BlockTreeItemImpl.java]: Let's do `Objects.hash(getTitle().getText(), getBlockChildren())`?

File: src/test/java/com/notably/storage/JsonBlockStorageTest.java [hunk 4]
```diff
@@ -0,0 +1,109 @@
...
+    /**
+     * Saves the {@code BlockTree} at the specified path.
+     *
+     * @param blockTree
+     * @param filePath
+     */
+    private void saveBlockTree(BlockTree blockTree, String filePath) {
+        try {
+            new JsonBlockStorage(Paths.get(filePath))
+                    .saveBlockTree(blockTree, addToTestDataPathIfNotNull(filePath));
```
Comment (kevinputera) [inline src/test/java/com/notably/storage/JsonBlockStorageTest.java:50]: Not sure if I'm getting this correctly, but I believe `new JsonBlockStorage(Paths.get(filePath)).saveBlockTree(blockTree);` would suffice?

**Review by kevinputera — APPROVED** [review]
LGTM!

File: src/test/java/com/notably/storage/JsonBlockStorageTest.java (hunk 4 above)
Comment (firzanarmani) [reply src/test/java/com/notably/storage/JsonBlockStorageTest.java:50]: `addToTestDataPathIfNotNull(filePath)` creates a `Path` object from the `filePath` which is actually a String, so this is needed!

**Pull request merged**